packages/apollo-react-relay-duct-tape/src/hooks.ts

```diff
@@ -175,3 +176,86 @@ export function useSubscription<TSubscriptionPayload extends OperationType>(
     }
   }
 }
+
+interface IMutationCommitterOptions<TMutationPayload extends OperationType> {
+  variables?: TMutationPayload["variables"];
+  optimisticResponse?: Partial<TMutationPayload["response"]> | null;
+}
+
+type MutationCommiter<TMutationPayload extends OperationType> = (
+  options: IMutationCommitterOptions<TMutationPayload>
+) => Promise<{ errors?: Error[]; data?: TMutationPayload["response"] }>;
+
+/**
+ * Declare use of a mutation within component. Returns an array of a function and loading state boolean.
+ *
+ * Returned function can be called to perform the actual mutation with provided variables.
+ *
+ * @param mutation Mutation document
+ * @returns [commitMutationFn, isInFlight]
+ *
+ * commitMutationFn
+ * @param options.variables map of variables to pass to mutation
+ * @param options.optimisticResponse proposed response to apply to the store while mutation is in flight
+ * @returns A Promise to an object with either errors or/and the result data
```

Comment:
Not entirely according to spec, seeing as it’s a disposable in react-relay, but it also doesn’t seem like a stretch to make it a Promise there in the future too https://github.com/facebook/relay/issues/1822, so I’m ok with merging as-is.